File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int maximum = 0;
        int left = 0, right = heights.length - 1;
        while(left < right){
            int area = Math.min(heights[left], heights[right]) * (right - left);
            maximum = Math.max(area, maximum);
            if(heights[right] > heights[left]){
                left++;
            } else {
                right--;
            }
            
        }
        return maximum;
    }
}

File: Data Structures & Algorithms/insert-new-interval/submission-0.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        res = []

        for i in range(len(intervals)):
            if newInterval[1] < intervals[i][0]:#non-overlapping and on the left
                res.append(newInterval)
                return res + intervals[i:]
            
            elif newInterval[0] > intervals[i][1]:#non-overlapping and on the right
                res.append(intervals[i])
                

            else:#overlapping cases
                newInterval = min(newInterval[0], intervals[i][0]), max(newInterval[1], intervals[i][1])

        res.append(newInterval)#in case the return does not execute

        return res 